tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -386,6 +453,28 @@ private boolean prepare() {
       LOG.error("Exception while preparing AM", e);
       return false;
     }
+
+    // Write config file to HDFS
+    jobDir = new Path(tonyHistoryFolder + appIdString + "/");
+    try {
+      if (!fs.exists(jobDir)) {
+        fs.mkdirs(jobDir);
+        fs.setPermission(jobDir, new FsPermission((short) 0770));
+      }
+    } catch (IOException e) {
+      LOG.error("Failed to create " + jobDir.toString(), e);
+      return false;
+    }
+
+    Path configFile = new Path(jobDir,"config.xml");
+
+    try (FSDataOutputStream out = fs.create(configFile)) {
+      tonyConf.writeXml(out);
```

Comment:
Again, let's move this out into a separate method that clearly describes what's being done, like `writeConfigHistoryFile`.

Reply:
Fixed: https://github.com/linkedin/TonY/pull/79/commits/e489758601fce6c775bd5d11e0f97289e0664a81